src/main.rs

```diff
@@ -34,11 +34,10 @@ fn get_from_db(db_url: &str, room_id: Option<&str>) -> BTreeMap<i64, Entry> {
             main.id AS state_group,
             forwards.state_group AS next,
             backwards.prev_state_group AS prev,
-            e.event_id IS NOT NULL AS is_referenced
+            EXISTS (SELECT 1 FROM event_to_state_groups WHERE state_group = main.id) AS is_referenced
```

Comment:
```suggestion
            EXISTS (SELECT 1 FROM event_to_state_groups WHERE state_group = main.id LIMIT 1) AS is_referenced
```

Does that change things at all I wonder?

Reply:
I'd kinda expect the postgres optimiser to figure out they are the same, but :man_shrugging:

Reply:
query plan looks the same on postgres 9.5 at least (in other news: I should upgrade)